dask/array/image.py

```diff
@@ -45,9 +45,14 @@ def imread(filename, imread=None, preprocess=None):
     will be treated as individual chunks
     """
     imread = imread or sk_imread
-    filenames = sorted(glob(filename))
-    if not filenames:
-        raise ValueError("No files found under name %s" % filename)
+    if isinstance(filename, str):
+        filenames = sorted(glob(filename))
+        if not filenames:
+            raise ValueError("No files found under name %s" % filename)
+    elif isinstance(filename, str):
```

Comment:
Is any iterable allowed,? If so, it'd be more flexible to express this like. 
```python
from collections.abc import Iterable
```

```suggestion
    elif isinstance(filename, Iterable):
```